### PR TITLE
56 update code to match new api spec

### PR DESCRIPTION
**Pull request by bschroeter**
A ridiculous amount of work to address the new API spec. I've been assured that any endpoint changes going forward will be namespaced (i.e. api/v1, api/v2 etc.) to avoid lengthy re-implementation delays.

I took this opportunity to improve docstrings etc. other comments will be written in-situ.

Let me know if anything is unclear.

### CODE REVIEW

File: meorg_client/cli.py [hunk 1]
```diff
@@ -185,7 +177,23 @@ def file_detach_all(output_id: str):
...
     client = _get_client()
-    _ = _call(client.detach_all_files_from_model_output, id=output_id)
+    _ = _call(client.delete_all_files_from_model_output, id=output_id)
+    click.echo("SUCCESS")
+
+
+@click.command("delete")
+@click.argument("output_id")
+@click.argument("file_id")
+def file_delete(output_id: str, file_id: str):
```
Comment (bschroeter) [inline meorg_client/cli.py:187]: New deleting method.

File: meorg_client/cli.py [hunk 2]
```diff
@@ -286,9 +294,8 @@ def cli_analysis():
 # Add file commands
 cli_file.add_command(file_list)
 cli_file.add_command(file_upload)
-cli_file.add_command(file_attach)
-cli_file.add_command(file_detach_all)
-
+cli_file.add_command(file_delete)
```
Comment (bschroeter) [inline meorg_client/cli.py:297]: Added for CLI access.

File: meorg_client/client.py [hunk 3]
```diff
@@ -221,8 +221,8 @@ def logout(self):
     def _upload_files_parallel(
         self,
         files: Union[str, Path, list],
+        id: str,
```
Comment (bschroeter) [inline meorg_client/client.py:224]: `attach_to` is renamed to `id` everywhere now.

File: meorg_client/client.py [hunk 4]
```diff
@@ -248,16 +248,17 @@ def _upload_files_parallel(
         # Do the parallel upload
         responses = None
         responses = meop.parallelise(
-            self._upload_file, n, files=files, attach_to=attach_to, progress=progress
+            self._upload_file, n, filepath=files, id=id, progress=progress
```
Comment (bschroeter) [inline meorg_client/client.py:251]: Passing the id over to the parallelise method.

File: meorg_client/client.py [hunk 5]
```diff
@@ -288,40 +290,27 @@ def upload_files(
...
-                    file_id = response.get("data").get("files")[0].get("file")
-                    file_ids.append(file_id)
-
-                self.attach_files_to_model_output(id=attach_to, files=file_ids)
-
-        return mu.ensure_list(responses)
+        # return mu.ensure_list(responses)
+        return responses
 
     def _upload_file(
```
Comment (bschroeter) [inline meorg_client/client.py:303]: `_upload_files` was capable of uploading multiple files in the one payload, however, this was causing issues with return types (lists vs single objects) so it was simpler to make it the singular case. This is fine as the packet limit on the upload basically prevents uploading multiple files of any meaningful size anyway.

TL;DR - making this single-file made everything a lot simpler for the other methods which all call this.

File: meorg_client/client.py [hunk 6]
```diff
@@ -400,42 +375,29 @@ def list_files(self, id: str) -> Union[dict, requests.Response]:
             method=mcc.HTTP_GET, endpoint=endpoints.FILE_LIST, url_params=dict(id=id)
         )
 
-    def attach_files_to_model_output(
-        self, id: str, files: list
-    ) -> Union[dict, requests.Response]:
-        """Attach files to a model output.
+    def delete_file_from_model_output(self, id: str, file_id: str):
```
Comment (bschroeter) [inline meorg_client/client.py:378]: New delete method, turns 2 calls into 1

File: meorg_client/endpoints.py [hunk 7]
```diff
@@ -9,9 +9,10 @@
 
 # Files
 FILE_LIST = "modeloutput/{id}/files"
-FILE_UPLOAD = "upload"
+FILE_UPLOAD = FILE_LIST
```
Comment (bschroeter) [inline meorg_client/endpoints.py:12]: new API endpoints

File: meorg_client/tests/test_cli.py [hunk 8]
```diff
@@ -1,10 +1,11 @@
+"""Test the CLI actions."""
```
Comment (bschroeter) [inline meorg_client/tests/test_cli.py:1]: Updated all of the tests to make things work. Cleaned up docstrings, switched out for fixtures.

**Review by ccarouge — APPROVED** [review]
LGTM. A simple typo-like comment since you were cleaning up the docstrings.
